cmd/tink-worker/container/container.go

```diff
@@ -0,0 +1,111 @@
+package container
+
+import (
+	"context"
+	"path"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+)
+
+type configData struct {
+	name           string
+	pid            string
+	env            []string
+	volumeBindings []string
+	cmd            []string
+	imageName      string
+	registry       string
+	captureLogs    bool
+}
+
+type removeOpts struct {
+	force   bool
+	links   bool
+	volumes bool
+}
+
+// create creates container and returns containerid on success else returns error.
+func create(ctx context.Context, cli client.ContainerAPIClient, data configData) (string, error) {
+	config := &container.Config{
+		Image:        path.Join(data.registry, data.imageName),
+		AttachStdout: true,
+		AttachStderr: true,
+		Cmd:          data.cmd,
+		Tty:          true,
+		Env:          data.env,
+	}
+	if !data.captureLogs {
+		config.AttachStdout = false
+		config.AttachStderr = false
+		config.Tty = false
+	}
+
+	hostConfig := &container.HostConfig{
+		Privileged: true,
+		Binds:      data.volumeBindings,
+	}
+
+	if pidConfig := data.pid; pidConfig != "" {
+		hostConfig.PidMode = container.PidMode(pidConfig)
+	}
+
+	resp, err := cli.ContainerCreate(ctx, config, hostConfig, nil, nil, data.name)
+	if err != nil {
+		return "", errors.Wrap(err, "DOCKER CREATE")
+	}
+	return resp.ID, nil
+}
+
+// start starts container and returns error on failure.
+func start(ctx context.Context, cli client.ContainerAPIClient, id string) error {
+	err := cli.ContainerStart(ctx, id, types.ContainerStartOptions{})
+	if err != nil {
+		return errors.Wrap(err, "DOCKER START")
+	}
+	return nil
+}
+
+// remove delete container and remove volumes, links if flags are set to true. returns error on failure.
+func remove(ctx context.Context, cli client.ContainerAPIClient, id string, opts removeOpts) error {
+	// create options for removing container
+	options := types.ContainerRemoveOptions{
+		Force:         opts.force,
+		RemoveLinks:   opts.links,
+		RemoveVolumes: opts.volumes,
+	}
+
+	// send API call to remove the container
+	err := cli.ContainerRemove(ctx, id, options)
```

Comment:
testing this I found that if a canceled context is passed in then `cli.ContainerRemove(ctx, id, options)` will fail and not remove the container. I was not expecting this. I needed to create a new context with a deadline, for example, and pass that in to make sure the container was removed. 

we might want to think about ways to help the user not run into this. On idea could be to use a new context with a deadline inside of remove. thoughts?

```Go
var cancel context.CancelFunc
select {
case <-ctx.Done():
	ctx, cancel = context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()
default:
}
```